src/lib.rs

```diff
@@ -216,6 +242,65 @@ impl Default for CertificateParams {
 	}
 }
 
+impl CertificateParams {
+	/// Parses the ca certificate from the ASCII PEM format
+	/// 
+	/// See `from_ca_cert_der` for more details.
+	#[cfg(all(feature = "pem", feature = "x509-parser"))]
+	pub fn from_ca_cert_pem(pem_str :&str, key_pair: KeyPair) -> Result<Self, RcgenError> {
+		let certificate = pem::parse(pem_str)
+			.or(Err(RcgenError::CouldNotParseCertificate))?;
+		let certificate_der :&[_] = &certificate.contents;
+		Self::from_ca_cert_der(certificate_der, key_pair)
+	}
+
+	/// Parses the ca certificate from the DER format
+	/// 
+	/// This function is only of use if you have an existing ca certificate with which
+	/// you want to sign a certificate newly generated by `rcgen` using the
+	/// `serialize_der_with_signer()` or `serialize_pem_with_signer()` functions.
+	/// 
+	/// Will not check if certificate is a ca certificate!
+	#[cfg(feature = "x509-parser")]
+	pub fn from_ca_cert_der(ca_cert: &[u8], key_pair: KeyPair) -> Result<Self, RcgenError> {
+		let (_remainder, x509) = x509_parser::parse_x509_der(ca_cert)
+			.or(Err(RcgenError::CouldNotParseCertificate))?;
+
+		// WDYT?
+		// if !x509.tbs_certificate.is_ca() {
+		// 	return Err(RcgenError::NoCaCertificate));
+		// }
+
+		let alg = SignatureAlgorithm::from_oid(x509.signature_algorithm.algorithm.iter().as_slice())?;
+
+		let mut dn = DistinguishedName::new();
+		for rdn in x509.tbs_certificate.subject.rdn_seq.iter() {
+			let attr = if rdn.set.len() > 1 {
+				// panic!("rcgen does not support distinguished names with more than one attribute");
+				return Err(RcgenError::CouldNotParseCertificate);
+			} else {
+				&rdn.set.as_slice()[0]
+			};
+			let value = attr.attr_value.as_slice()
+				.or(Err(RcgenError::CouldNotParseCertificate))?;
+
+			let dn_type = DnType::from_oid(attr.attr_type.iter().as_slice());
+			let dn_value = String::from_utf8(value.into())
+				.or(Err(RcgenError::CouldNotParseCertificate))?;
+			dn.push(dn_type, dn_value);
+		}
+
+		Ok(
+			CertificateParams {
+				alg: alg,
```

Comment:
style nit: please put space before and after the colon.

Reply:
Done